diff --git a/main/main.c b/main/main.c
index 6536a216..bacfaa1b 100644
--- a/main/main.c
+++ b/main/main.c
@@ -22,9 +22,9 @@
 */
 #include "general.h"  /* must always come first */
 
-#if HAVE_DECL___ENVIRON
+#ifdef HAVE_DECL___ENVIRON
 #include <unistd.h>
-#elif HAVE_DECL__NSGETENVIRON
+#elif defined (HAVE_DECL__NSGETENVIRON)
 #include <crt_externs.h>
 #endif
 
@@ -508,9 +508,9 @@ static void sanitizeEnviron (void)
 	char **e;
 	int i;
 
-#if HAVE_DECL___ENVIRON
+#ifdef HAVE_DECL___ENVIRON
 	e = __environ;
-#elif HAVE_DECL__NSGETENVIRON
+#elif defined (HAVE_DECL__NSGETENVIRON)
 	{
 		char ***ep = _NSGetEnviron();
 		if (ep)
